mqtt/src/test/scala/akka/stream/alpakka/mqtt/scaladsl/MqttSourceSpec.scala

```diff
@@ -169,5 +170,40 @@ class MqttSourceSpec
         elem2.futureValue shouldBe MqttMessage(topic1, ByteString("ohi"))
       }
     }
+
+    "support will message" in {
+      import system.dispatcher
+
+      val ks = Tcp()
+        .bind("localhost", 1337)
+        .runWith(Sink.head)
+        .map(
+          _.handleWith(Tcp().outgoingConnection("localhost", 1883).viaMat(KillSwitches.single)(Keep.right))
+        )
+
+      {
+        val settings = MqttSourceSettings(
+            sourceSettings
+              .withBroker("tcp://localhost:1337")
+              .withWill(Will(MqttMessage(willTopic, ByteString("ohi")), MqttQoS.AtLeastOnce, retained = true)),
```

Comment:
Use different client ids for both sources. For example `testator` and `executor`.